src/app/clusters/operational-credentials/operational-credentials.cpp

```diff
@@ -33,33 +36,193 @@
+    }
+};
+
+OpCredsAdminPairingTableDelegate gAdminDelegate;
+
+void emberAfPluginOperationalCredentialsServerInitCallback(void)
+{
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: Initiating OpCreds cluster by writing fabrics list from admin pairing table.");
+    GetGlobalAdminPairingTable().SetAdminPairingDelegate(&gAdminDelegate);
+    writeAdminsIntoFabricsListAttribute();
+}
 
 bool emberAfOperationalCredentialsClusterRemoveFabricCallback(chip::app::Command * commandObj, chip::FabricId fabricId,
                                                               chip::NodeId nodeId, uint16_t vendorId)
 {
-    // Go through admin pairing table and remove the element
-    // Then fetch the attribute list and find the right element (or maybe just use the admin pairing table index)
-    // Then rewrite the entire attribute list (no choice for now)
     emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: RemoveFabric"); // TODO: Generate emberAfFabricClusterPrintln
-    EmberAfStatus status = EMBER_ZCL_STATUS_FAILURE;
+    
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    AdminPairingInfo * admin;
+    AdminId adminId;
+    CHIP_ERROR err;
+
+    // Fetch current admin
+    admin = retrieveCurrentAdmin();
+    VerifyOrExit(admin != nullptr, status = EMBER_ZCL_STATUS_SUCCESS); // Admin has already been removed
+
+    // Delete admin
+    adminId = admin->GetAdminId();
+    GetGlobalAdminPairingTable().ReleaseAdminId(adminId);
+    err = AdminPairingInfo::DeleteFromKVS(adminId);
+    VerifyOrExit(err == CHIP_NO_ERROR, status = EMBER_ZCL_STATUS_FAILURE);
+
+exit:
+
     emberAfSendImmediateDefaultResponse(status);
     return true;
 }
 
-bool emberAfOperationalCredentialsClusterGetFabricIdCallback(chip::app::Command * commandObj)
+bool emberAfOperationalCredentialsClusterSetFabricCallback(chip::app::Command * commandObj, uint16_t VendorId)
 {
-    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: GetFabricId");
-    EmberAfStatus status = EMBER_ZCL_STATUS_FAILURE;
-    emberAfSendImmediateDefaultResponse(status);
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: SetFabric with vendorId %d", VendorId);
+
+    EmberAfStatus status = EMBER_ZCL_STATUS_SUCCESS;
+    EmberStatus sendStatus = EMBER_SUCCESS;
+    CHIP_ERROR err;
+
+    // Fetch current admin
+    AdminPairingInfo * admin = retrieveCurrentAdmin();
+    VerifyOrExit(admin != nullptr, status = EMBER_ZCL_STATUS_FAILURE);
+
+    // Store vendorId
+    admin->SetVendorId(VendorId);
+    emberAfPrintln(EMBER_AF_PRINT_DEBUG, "OpCreds: vendorId is now set %d", admin->GetVendorId());
+    err = admin->StoreIntoKVS();
+    VerifyOrExit(err == CHIP_NO_ERROR, status = EMBER_ZCL_STATUS_FAILURE);
+
+    // Return FabricId
+    emberAfFillExternalBuffer((ZCL_CLUSTER_SPECIFIC_COMMAND | ZCL_FRAME_CONTROL_SERVER_TO_CLIENT), ZCL_OPERATIONAL_CREDENTIALS_CLUSTER_ID,
+                              ZCL_SET_FABRIC_RESPONSE_COMMAND_ID, "y", emberAfCurrentCommand()->source);
```

Comment:
Again, the "source" of a command is not a fabric id.

Reply:
that is what I am using for now. I had explained it in my previous patch.

Reply:
Again, needs documentation or pointer to documentation, and definitely pointer to issue.